File: apps/api/FinanceControl.Shared/Dtos/Request/CreateWishlistItemRequestDto.cs
using FinanceControl.Shared.Enums;

namespace FinanceControl.Shared.Dtos.Request
{
    public class CreateWishlistItemRequestDto
    {
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public int? TargetPrice { get; set; }
        public EnumWishlistPriority Priority { get; set; } = EnumWishlistPriority.Medium;
        public string? Url { get; set; }
        public string? ImageUrl { get; set; }
    }
}

File: apps/mobile/lib/features/wishlist/data/dtos/create_wishlist_item_request_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'create_wishlist_item_request_dto.g.dart';

@JsonSerializable(createFactory: false, includeIfNull: false)
class CreateWishlistItemRequestDto {
  const CreateWishlistItemRequestDto({
    required this.name,
    this.description,
    this.targetPrice,
    this.priority,
    this.url,
    this.imageUrl,
  });

  final String name;
  final String? description;
  final int? targetPrice;
  final String? priority;
  final String? url;
  final String? imageUrl;

  Map<String, dynamic> toJson() => _$CreateWishlistItemRequestDtoToJson(this);
}
